specs/BackgroundTask/BackgroundTaskBuilder.md

```diff
@@ -0,0 +1,98 @@
+This API takes care of avoiding this workaround so that WinAppSDK applications can directly register the full trust COM components with the background tasks.
+
+# Conceptual pages (How To)
+
+ ## Register Background Task (c++)
+
+ Registering a Background Task for full trust COM component. Developers may also need to set the UniversalBGTask property as true in their project configuration.
+
+ ```c++
+       //Using WinAppSDK API for BackgroundTaskBuilder
+        winrt::Microsoft::Windows::ApplicationModel::Background::BackgroundTaskBuilder builder;
+        SystemTrigger trigger = SystemTrigger(SystemTriggerType::TimeZoneChange, false);
+        auto backgroundTrigger = trigger.as<IBackgroundTrigger>();
+        builder.SetTrigger(backgroundTrigger);
+        builder.SetTaskEntryPointClsid(classGuid);
+        builder.Register(); 
+```
+
+# API Pages
+
+## BackgroundTaskBuilder class
+
+Represents a background task to register with the system.
+
+This class is not agile, which means that you need to consider its threading model and marshaling behavior. For more info, see [Threading and Marshaling (C++/CX)]( https://learn.microsoft.com/en-us/cpp/cppcx/threading-and-marshaling-c-cx?view=msvc-170 ) and [Using Windows Runtime objects in a multithreaded environment (.NET)]( https://learn.microsoft.com/en-us/windows/uwp/threading-async/using-windows-runtime-objects-in-a-multithreaded-environment)
+
+
+## BackgroundTaskBuilder constructor
+
+| Name | Description |
+|-|-|
+| BackgroundTaskBuilder  | Create a new BackgroundTaskBuilder object.|
+
+
+## BackgroundTaskBuilder Properties
+
+| Name | Description | Value |
+|-|-|-|
+|Name| Gets or sets the name of a background task.|Name(A string representing name of the task) |
+|TaskGroup| Gets and sets the group identifier.| TaskGroup (The group identifier.) |
+
+
+## BackgroundTaskBuilder Methods
+
+| Name | Description | Parameters | Returns |
+|-|-|-|-|
+| AddCondition(IBackgroundCondition) | Adds a condition to a background task.| __IBackgroundCondition__ Condition for Background Task | |
+| Register() | Registers a background task with the system.| | __BackgroundTaskRegistration__|
+
+
+
+# API Details
+
+> Note: all of this new WinAppSDK API is the same as the existing UWP API
+ [BackgroundTaskBuilder](https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.background.backgroundtaskbuilder), only changing the register operation internally to use WindowsRuntimeComponent for registration and then use the WindowsRuntimeComponent to activate the COM component of the application.
+```c++ (but really MIDL3)
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+namespace Microsoft.Windows.ApplicationModel.Background
+{
+    [contractversion(1)]
+    apicontract BackgroundTaskContract{};
+
+    [contract(BackgroundTaskContract, 1)]
+    runtimeclass BackgroundTaskBuilder
+    {
+        BackgroundTaskBuilder();
+
+        void SetTrigger(
```

Comment:
Why SetTrigger and SetTaskEntryPointClsId functions are not listed in the above BackgroundTaskBuilder Methods table.

Reply:
all good here, I read the notes afterwards.

Reply:
Added both, Thanks